service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AuthenticationMessageCallback.java

```diff
@@ -0,0 +1,20 @@
+package com.microsoft.azure.sdk.iot.service.transport.amqps;
+
+import org.apache.qpid.proton.amqp.transport.DeliveryState;
+import org.apache.qpid.proton.reactor.Reactor;
+
+/**
+ * Interface that defines the callback used by the CBS receiver link to notify the CBS session when an authentication
+ * message has been received from the service
+ */
+interface AuthenticationMessageCallback
+{
+    /**
+     * This callback executes when the CBS receiver link receives an authentication message from the service. This
+     * message contains a status code and a description.
+     * @param status The status of the authentication process (200 is successful).
+     * @param description The human readable explanation for the status code.
+     * @return The AMQP acknowledgement type for the receiver link to acknowledge the authentication message with.
```

Comment:
does this always follow a request - response model? what happens if we do not send the ack? (eg, if the device got disconnected before it could send the ack)?

Reply:
From what I remember, IoY Hub doesn't actually care about this ack. I still think it is worth it to send it though